pkgs/by-name/po/poco/package.nix

```diff
@@ -54,25 +55,54 @@ stdenv.mkDerivation rec {
   MYSQL_DIR = libmysqlclient;
   MYSQL_INCLUDE_DIR = "${MYSQL_DIR}/include/mysql";
 
-  cmakeFlags = [
-    # use nix provided versions of sqlite, zlib, pcre, expat, ... instead of bundled versions
-    (lib.cmakeBool "POCO_UNBUNDLED" true)
-  ];
+  cmakeFlags =
+    let
+      excludeTestsRegex = lib.concatStringsSep "|" [
+        # These tests require running services, which the checkPhase is ill equipeed to provide
+        # TODO get them running in a nixosTest
+        "Redis"
+        "DataODBC"
+        "MongoDB"
+        "DataMySQL"
+        # network not accessible from nix sandbox
+        "NetSSL" # around 25 test failures
+        "Net" # could be made to work when public network access is patched out
+      ];
+    in
+    [
+      # use nix provided versions of sqlite, zlib, pcre, expat, ... instead of bundled versions
+      (lib.cmakeBool "POCO_UNBUNDLED" true)
+      (lib.cmakeBool "ENABLE_TESTS" true)
+      (lib.cmakeFeature "CMAKE_CTEST_ARGUMENTS" "--exclude-regex;'${excludeTestsRegex}'")
+    ];
+
+  patches =
+    [
+      # Remove on next release
+      (fetchpatch {
+        name = "disable-included-pcre-if-pcre-is-linked-staticly";
+        # this happens when building pkgsStatic.poco
+        url = "https://patch-diff.githubusercontent.com/raw/pocoproject/poco/pull/4879.patch";
+        hash = "sha256-VFWuRuf0GPYFp43WKI8utl+agP+7a5biLg7m64EMnVo=";
+      })
+    ]
+    ++ lib.optionals stdenv.hostPlatform.isDarwin [
+      ./disable-broken-tests-darwin.patch
+    ]
+    ++ lib.optionals stdenv.hostPlatform.isLinux [
+      ./disable-broken-tests-linux.patch
+    ];
 
-  patches = [
-    # Remove on next release
-    (fetchpatch {
-      name = "disable-included-pcre-if-pcre-is-linked-staticly";
-      # this happens when building pkgsStatic.poco
-      url = "https://patch-diff.githubusercontent.com/raw/pocoproject/poco/pull/4879.patch";
-      hash = "sha256-VFWuRuf0GPYFp43WKI8utl+agP+7a5biLg7m64EMnVo=";
-    })
+  doCheck = true;
+  nativeCheckInputs = [
+    # workaround for some tests trying to write to /homeless-shelter
+    writableTmpDirAsHomeHook
   ];
 
   postFixup = ''
     grep -rlF INTERFACE_INCLUDE_DIRECTORIES "$dev/lib/cmake/Poco" | while read -r f; do
       substituteInPlace "$f" \
-        --replace "$"'{_IMPORT_PREFIX}/include' ""
+        --replace-quiet "$"'{_IMPORT_PREFIX}/include' ""
```

Comment:
Why not `--replace-fail` ?

Reply:
I tried, but the way the replace is phrased, it tries the replace on many files that do not contain the replace string. I thought it better to at least get rid of the warning this way.

Reply:
Perhaps we could identify those files and filter them? I don't think using replace-quiet is a good practice.

Reply:
:/ I'm not loving it.

Something like this would be possible

```
grep -rlF INTERFACE_INCLUDE_DIRECTORIES "$dev/lib/cmake/Poco" | \
while read -r f; do
  if grep -qF '${_IMPORT_PREFIX}/include' "$f"; then
    substituteInPlace "$f" \
      --replace-fail '$'"{_IMPORT_PREFIX}/include" ""
  fi
done
```

But to me, I don't really see what its adding over `--replace-quiet`

Reply:
Sadly the original commit the added this, doesn't really say what and why this is necessary, so I'm kind of lost as to what this is doing and why it is necessary.